src/Logging/Logging/ScopedLogService.cs

```diff
@@ -0,0 +1,84 @@
+﻿using CodeMonkeys.Configuration;
+
+using System;
+
+namespace CodeMonkeys.Logging
+{
+    public abstract class ScopedLogService<TOptions> : OptionsConsumer<TOptions>, IScopedLogService
+        where TOptions : LogOptions, new()
+    {
+        /// <summary>
+        /// Flag which indicates if the service accepts and queues writes.
+        /// <para>Defaults to <see langword="true"/>.</para>
+        /// </summary>
+        public virtual bool IsEnabled { get; protected set; } = true;
+
+        protected string Context { get; private set; }
+        protected LogMessageFormatter MessageFormatter { get; set; }
+
+        protected ScopedLogService(string context)
+        {
+            Context = context;
+            MessageFormatter = new LogMessageFormatter();
+        }
+
+        public bool IsEnabledFor(LogLevel logLevel)
+        {
+            if (IsEnabled)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/gh/UltimateCodeMonkeys/CodeMonkeysMVVM/pullRequest?prid=5626772)